### ISSUE TITLE
Limited scope ignore

### ISSUE THREAD

**Issue by GoMartin300** [streetsidesoftware/vscode-spell-checker#465]
I'm trying to get vscode-spell-checker to ignore some words only within a limited scope within my project as I don't want to ignore these words in my entire project.
Currently I am partially able to do this using:
```
// cspell:disable-line -- disables checking for the current line.
// cspell:disable-next-line -- disables checking till the end of the next line.
```
As mentioned in https://github.com/streetsidesoftware/vscode-spell-checker/issues/277#issuecomment-437594307
This works as expected in Latex (my use case):
```
kookies raises a warning on this line.
kookies does not raise a warning on this line.  % cspell:disable-line
```
But it falls a bit short when you consider that a single Latex line might contain an entire subsection, meaning that an entire subsection would be unchecked.
It also doesn't allow users to ignore words within multi-line listings such as:
```
\begin{lstlisting}
    kookies
    caikes
    cocolate
\end{lstlisting}
```
Any kind of comment in there would show up in the final document.

Both problems could probably be solved by adding the ability to ignore words only within a single file. I think it's acceptable to ignore "kookies" within an entire file, but not to ignore it within an entire project.

Ideally for my case, there would also be an ability to ignore words a single time on a single line. But that might not add that much value.

**Comment by GoMartin300** [comment]
This is already possible:
https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker

**Comment by villelahdenvuo** [comment]
This is also a problem for JavaScript template literals, it's difficult to ignore a multiline literal since you cannot add comments in the middle of the string easily.

**Comment by Jason3S** [comment]
Why not do it like this?

```ts
// cspell:disable
export const FOREIGN_VAT_NOTE = oneLine`Tilaus toimitetaan ulkomaille, tarkista arvonlisäveron määrä.
	Lisää EU:n VAT "lisätietoja jakeluihin" kohdasta.`;
// cspell:enable
```

By the way, adding a comment like:
```ts
// cspell:ignore Tilaus toimitetaan ulkomaille, tarkista arvonlisäveron määrä.
// cspell:ignore Lisää lisätietoja jakeluihin kohdasta
export const FOREIGN_VAT_NOTE = oneLine`Tilaus toimitetaan ulkomaille, tarkista arvonlisäveron määrä.
	Lisää EU:n VAT "lisätietoja jakeluihin" kohdasta.`;
```

is local to the file, not your entire project.
